lib/git.zsh

```diff
@@ -164,8 +168,46 @@ function git_prompt_long_sha() {
   SHA=$(__git_prompt_git rev-parse HEAD 2> /dev/null) && echo "$ZSH_THEME_GIT_PROMPT_SHA_BEFORE$SHA$ZSH_THEME_GIT_PROMPT_SHA_AFTER"
 }
 
+function git_repo_files_number() {
+  local gitdir
+  local indexf='index'
+  local X
+  local N
+  #N=$(__git_prompt_git ls-files 2> /dev/null | wc -l) # `git ls-files` is much faster than `git status` but still too slow!
+  gitdir=$(__git_prompt_git rev-parse --git-dir 2> /dev/null)
+  [[ $? -eq 128 ]] && return 128
+  [[ "$(__git_prompt_git config --get oh-my-zsh.hide-info 2>/dev/null)" == 1 ]] && return 0
+  # parse index file(s) [https://git-scm.com/docs/index-format]
+  X=$(xxd -p -l4 -s12 "$gitdir/$indexf" 2>/dev/null) # extension signature
+  [[ $? -ne 0 ]] && return 2
+  if [[ "$X" == "6c696e6b" ]]; then # "link" extension --> split/shared index
```

Comment:
Might be because I'm on my phone but why the specific SHA? Thanks for clarifying.

Reply:
```
# parse index file(s) [https://git-scm.com/docs/index-format]
  X=$(xxd -p -l4 -s12 "$gitdir/$indexf" 2>/dev/null) # extension signature
  [[ $? -ne 0 ]] && return 2
  if [[ "$X" == "6c696e6b" ]]; then # "link" extension --> split/shared index
```
As shown in [index-format](https://git-scm.com/docs/index-format):

- After the 12-byte header is a 4-byte extension signature, hence `X=$(xxd -p -l4 -s12 ...)`
- The [split index](https://git-scm.com/docs/index-format#_split_index) has an extension signature literally spelled "`link`" in ASCII whose HEX value is `0x6c696e6b`, hence `[[ "$X" == "6c696e6b" ]]`. We can use `xxd -r` to convert it back from HEX to ASCII, but it's obviously superfluous.